prow/github/types_test.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package github
+
+import (
+	"testing"
+)
+
+const capsTestLogin string = "BenTheElder"
+
+func TestIssueCapsLogin(t *testing.T) {
+	issue := Issue{
+		User: User{
+			Login: capsTestLogin,
+		},
+		Assignees: []User{
+			{
+				Login: capsTestLogin,
+			},
+		},
+	}
+
+	normalized := NormLogin(capsTestLogin)
+
+	if !issue.IsAuthor(capsTestLogin) {
+		t.Errorf("expected issue.IsAuthor(%s) to be true", capsTestLogin)
+	}
+	if !issue.IsAuthor(normalized) {
+		t.Errorf("expected issue.IsAuthor(%s) to be true", normalized)
+	}
+
+	if !issue.IsAssignee(capsTestLogin) {
+		t.Errorf("expected issue.IsAssignee(%s) to be true", capsTestLogin)
+	}
+	if !issue.IsAssignee(normalized) {
+		t.Errorf("expected issue.IsAssignee(%s) to be true", normalized)
+	}
```

Comment:
Can you add some test cases for casing besides the original and normalized? Those are the cases that are most interesting. For example `BENTHEELDER` and `BeNtHeElDeR` should both work for `IsAuthor` and `IsAssignee`

Reply:
Done, I added a set of valid alternatives + an explicitly normalized entry.